### PR TITLE
Added GoDoc for functions mentioned in Task 7 (#14004) team_store.go

### PR DESCRIPTION
**Pull request by palcodes**
#### Summary

This PR adds GoDoc for the functions mentioned in #14004 

* Fixes -> https://github.com/mattermost/mattermost-server/issues/14004
* JIRA ticket -> https://mattermost.atlassian.net/browse/MM-23002

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/team_store.go [hunk 1]
```diff
@@ -1022,6 +1023,7 @@ func (s SqlTeamStore) GetMember(teamId string, userId string) (*model.TeamMember
 	return dbMember.ToModel(), nil
 }
 
+// GetMembers returns a list of members from the database that match the teamId passed as parameter
```
Comment (jespino) [inline store/sqlstore/team_store.go:1026]: I would expect some extra information here around the `teamMemberGetOoptions` parameter, something like "the teamMembersGetOptions parameter allow you to restrict what is visible to the user, sort the results or exclude deleted users".

File: store/sqlstore/team_store.go [hunk 2]
```diff
@@ -1063,6 +1065,7 @@ func (s SqlTeamStore) GetMembers(teamId string, offset int, limit int, teamMembe
 	return dbMembers.ToModel(), nil
 }
 
+// GetTotalMemberCount returns the number of all members in a team for the teamId passed as a parameter.
```
Comment (jespino) [inline store/sqlstore/team_store.go:1068]: I would expect here to see something about the `restrictions` parameter. Something like "The restrictions parameters allows you to define the set of channels and teams that are visible to the caller of the query, and applies that restrictions to the results."

I would expect something similar in the other places where `restrictions` is passed as parameter.

**Review by jespino — CHANGES_REQUESTED** [review]
It is looking great, but i would love to have a bit more info about the most complex parameters passed in the functions. Thanks @palcodes!

**Comment by palcodes** [comment]
@jespino yea, that makes sense. I will get to it.

**Review by jespino — APPROVED** [review]
Now looks good to me. Thanks @palcodes! :confetti_ball:

**Comment by palcodes** [comment]
Hey, thanks for approving. Meet u on my next contribution 🤗

File: store/sqlstore/team_store.go [hunk 3]
```diff
@@ -1000,6 +1000,7 @@ func (s SqlTeamStore) UpdateMember(member *model.TeamMember) (*model.TeamMember,
 	return members[0], nil
 }
 
+// GetMember returns a single member of the team that matches the teamId and userId provided as parameters
```
Comment (justinegeffen) [inline store/sqlstore/team_store.go:1003]:
```suggestion
// GetMember returns a single member of the team that matches the teamId and userId provided as parameters.
```

File: store/sqlstore/team_store.go [hunk 4]
```diff
@@ -1107,6 +1119,8 @@ func (s SqlTeamStore) GetActiveMemberCount(teamId string, restrictions *model.Vi
 	return count, nil
 }
 
+// GetMembersByIds returns a list of members from the database that matches the teamId and the list of userIds passed as parameters
```
Comment (justinegeffen) [inline store/sqlstore/team_store.go:1122]:
```suggestion
// GetMembersByIds returns a list of members from the database that matches the teamId and the list of userIds passed as parameters.
```

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Just one comment from me, otherwise looks awesome. Thank you!

**Comment by palcodes** [comment]
I have made the required punctuation changes, what's the current status of the PR?

**Comment by hanzei** [comment]
The PR is good to go, but the pipeline fails for unknown reasons. Let me look into it.

**Comment by hanzei** [comment]
Finally CI is green. Sorry it took so long. Thanks for the PR @palcodes :+1:

**Pull request merged**